src/assets/RestApi/rest-api.md

```diff
@@ -0,0 +1,400 @@
+Example:
+```bash
+curl http://127.0.0.1:5000/tasks/1
+```
+The `curl` command sends an HTTP GET request to `http://127.0.0.1:5000/tasks/1`, retrieving the task with ID `1` from a local server.
+
+Expected Result:
+```json
+{
+  "task": {
+    "completed": false,
+    "id": 1,
+    "title": "Buy groceries"
+  }
+}
+```
+
+### 3. Create a New Task
+To create a new task, you can use following cURL command, replacing `<task_data>` with the JSON data for the new task:
+```bash
+curl -X POST -H "Content-Type: application/json" -d '<json_data>' http://127.0.0.1:5000/tasks
+```
+Example:
+```bash
+curl -X POST -H "Content-Type: application/json" -d '{"id":3,"task":"Finish the report","completed":false}' http://127.0.0.1:5000/tasks
+```
+The `curl` command sends an HTTP POST request with JSON data to `http://127.0.0.1:5000/tasks`, creating a new task.
+
+Expected Result:
+```json
+{
+  "task": {
+    "completed": false,
+    "id": 3,
+    "task": "Finish the report"
+  }
+}
+```
+
+### 4. Update a Task
+To update an existing task, use the following cURL command, replacing `<task_id>` with the id of the task, and `<updated_data>` with the JSON data you want to update:
+```bash
+curl -X PUT -H "Content-Type: application/json" -d '<updated_data>' http://127.0.0.1:5000/tasks/<task_id>
+```
+Example:
+```bash
+curl -X PUT -H "Content-Type: application/json" -d '{"completed":true}' http://127.0.0.1:5000/tasks/3
+```
+The `curl` command sends an HTTP PUT request with JSON data to `http://127.0.0.1:5000/tasks/3`, updating the task with ID `3` to mark it as completed.
+
+Expected Result:
+```json
+{
+  "task": {
+    "completed": true,
+    "id": 3,
+    "task": "Finish the report"
+  }
+}
+```
+
+### 5. Delete a Task
+To delete a task, use the following cURL command, replacing `<task_id>` with the id of the task you want to delete:
+```bash
+curl -X DELETE http://127.0.0.1:5000/tasks/<task_id>
+```
+Example:
+```bash
+curl -X DELETE http://127.0.0.1:5000/tasks/3
+```
+The `curl` command sends an HTTP DELETE request to `http://127.0.0.1:5000/tasks/3`, deleting the task with ID `3`.
+
+Expected Result:
+```json
+{
+  "result": "Task deleted"
+}
+```
+
+Let's try to retrieve a task that doesn't exist in the list of tasks. For instance, we just deleted a task with id = 3. Now, let's try to retrieve the task with id = 3 and see the result.
+```bash
+curl http://127.0.0.1:5000/tasks/3
+```
+Expected result:
+```json
+{
+  "error": "Task not found"
+}
+```
+We see that the JSON response returns the error `Task not found`, which indicates that the task doesn't exist in the list of tasks.
+
+## Common Causes of API Request Failures
+Now, let's explore scenarios where API requests might fail. API requests can fail due to various reasons, including sending invalid requests. This could happen if an incorrect endpoint is used, if the JSON data is invalid, or if the request is made with an unsupported HTTP method. Let's examine some instances where API requests might fail.
```

Comment:
```suggestion
Now, let's explore scenarios where API requests might fail. API requests can fail for various reasons, including sending invalid requests. This could happen if an incorrect endpoint is used, if the JSON data is invalid, or if the request is made with an unsupported HTTP method. Let's examine some instances where API requests might fail.
```